SpatialGDK/Source/SpatialGDK/Private/Utils/ActorGroupManager.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Utils/ActorGroupManager.h"
+#include "SpatialGDKSettings.h"
+
+void UActorGroupManager::Init()
+{
+	if (const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>())
+	{
+		DefaultWorkerType = Settings->DefaultWorkerType.WorkerTypeName;
+		if (Settings->bEnableOffloading)
+		{
+			for (TPair<FName, FActorGroupInfo> ActorGroup : Settings->ActorGroups)
+			{
+				ActorGroupToWorkerType.Add(ActorGroup.Key, ActorGroup.Value.OwningWorkerType.WorkerTypeName);
+
+				for (TSoftClassPtr<AActor> ClassPtr : ActorGroup.Value.ActorClasses)
+				{
+					ClassPathToActorGroup.Add(ClassPtr, ActorGroup.Key);
+				}
+			}
+		}
+	}
+}
+
+FName UActorGroupManager::GetActorGroupForClass(TSubclassOf<AActor> Class)
+{
+	if (Class == nullptr)
+	{
+		return NAME_None;
+	}
+
+	UClass* FoundClass = Class;
+	TSoftClassPtr<AActor> ClassPtr = TSoftClassPtr<AActor>(FoundClass);
```

Comment:
Think this especially could do with documentation as to it's usefulness, which is pretty obscure to the average coder 😄

Reply:
What you explained to me about not loading all the classes

Reply:
Added explanation in comment in FActorGroup